### ISSUE TITLE
Some tasks are being skipped (pagination needed for +100 pr comments)

### ISSUE THREAD

**Issue by billsacks** [stilliard/github-task-list-completed#19]
I have been using this app for about 8 months now and it has been working great – thank you!

Today, for the first time, I noticed that it is skipping (not counting) some tasks in one of our PRs: https://github.com/ESCOMP/CTSM/pull/1301. Two check boxes that seem to be skipped are the ones in https://github.com/ESCOMP/CTSM/pull/1301#discussion_r614995519 and https://github.com/ESCOMP/CTSM/pull/1301#discussion_r614998877 – checking and unchecking those has no effect on the summary count, even though others in the PR (like https://github.com/ESCOMP/CTSM/pull/1301#pullrequestreview-630252567) *are* counted. There seem to be a few others in the PR that also aren't counted.

I can't tell what's going on here: I tried making a new PR in a test repo to reproduce this problem and wasn't able to.

This isn't critical for me, since I tend to double-check the final count with my own tool (https://github.com/ESMCI/github-tools), so feel free to close as a wontfix if you also can't reproduce it – but I wanted to raise this in case it's a more general issue.

**Comment by stilliard** [comment]
Hey @billsacks !
Thanks, glad you're getting good use of the app still. 

Ah interesting, I wonder if it's an issue where maybe GitHub's only giving me a limited set of comments or similar? I'll check with a test repo and see if after a certain number of comments maybe it stops. 

Thanks for reporting the issue, will keep you posted as I debug.

**Comment by stilliard** [comment]
Interesting, in a test it stopped at 30 comments:
https://github.com/stilliard/github-task-list-completed/pull/20

Will be back online over the weekend so I'll check the API docs then as maybe GitHub has a default limit of 30 per page? 
Hopefully we can just bump that to 100 or more, or we could easily add in paging over the comments if needed.

**Comment by stilliard** [comment]
In fact, I've got a couple minutes before I need to disappear tonight. 
Found the limit for comments on PR's is in fact 30, ref: https://docs.github.com/en/rest/guides/traversing-with-pagination

I'll bump this with the `per_page` param to 100 for now but will also look to come back and add full paging support in too.

**Comment by billsacks** [comment]
Awesome, thanks a lot for looking into this!

**Comment by stilliard** [comment]
No problem, thanks for finding the bug! 

The fix for checking up to 100 of each type of comment on a PR is now merged and live, when you can could you let me know if that's resolved the issue you were seeing too or if there's still any issue, thanks. 

I'll keep the issue open for now to remind me to come back and add proper pagination in as well. 

Hope you have a great weekend :wave:

**Comment by billsacks** [comment]
Yes, the PR that had issues before now has a correct count – thanks again!

(We do sometimes have PRs that accumulate hundreds of comments, so long-term it would be awesome to have this pagination, but that's not at all critical.)
